db/migrate/20241023012600_add_apple_show_id_to_feeds.rb

```diff
@@ -0,0 +1,6 @@
+class AddAppleShowIdToFeeds < ActiveRecord::Migration[7.2]
+  def change
+    add_column :feeds, :apple_show_id, :string
```

Comment:
This is just an idea - I want to discuss and perhaps change this

Reply:
From what I gathered in discussion, this is such a commonly used external id that it makes sense to pull this up onto the feed model itself so that it can be used as an XID across many different services. +1

Reply:
We just need a way to override or replace the existing mechanism for linking a show to the external Apple resource.  The way it's done currently is with the SyncLog model. https://github.com/PRX/feeder.prx.org/blob/main/app/models/apple/show.rb#L108-L114

In the old init, the linking happened like so:
https://github.com/PRX/feeder.prx.org/blob/e35f2c7d29ae2266cdadd37ec74de2c18fad6a04/app/models/apple/config.rb#L93

In the current system, that initialization routine is set up under a rails hook:
https://github.com/PRX/feeder.prx.org/blob/main/app/models/feeds/apple_subscription.rb#L26-L36

`Apple::Show.connect_existing` could be a good interface to use? That, plus the associated changes in the `Show#apple_id`, could get you something.

Reply:
I tried that, but it always tries to insert, and when the value emptied, or updated, it throws an error.
I think I need to make it a bit more complex to handle those cases

Reply:
ok, I updated the `connect_existing` method to handle when the apple_show_id changes or is removed.